library/src/main/java/com/github/mproberts/rxdatabinding/bindings/RecyclerViewDataBindings.java

```diff
@@ -340,6 +344,30 @@ private RecyclerViewAdapter(ItemDataProvider dataProvider, ItemViewCreator viewC
         public void onAttachedToRecyclerView(RecyclerView recyclerView) {
             super.onAttachedToRecyclerView(recyclerView);
 
+            if (_lifecycle != null) {
+                _lifecycle.setActive(false); // will trigger a call to unsubscribe() if active
```

Comment:
I'm not sure if we actually need this. Technically if we skip it, the old lifecycle will never change. But if I bind a new adapter to a recycler view, I would expect the previous subscription to be disposed of and a new one to be created.

@mproberts let me know what you think.

Reply:
This will call through to unsubscribe and clear out the existing _subscription, which will allow the new lifecycle's actual activation to attach again. Since the _lifecycle isn't actually doing the subscription management and it's managed by instance state in the RecyclerViewAdapter I guess it's moot.